osu.Game/Rulesets/Judgements/JudgementResultEntry.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Game.Rulesets.Objects;
+
+namespace osu.Game.Rulesets.Judgements
+{
+    internal class JudgementResultEntry : IComparable<JudgementResultEntry>
+    {
+        public readonly double Time;
+
+        public readonly HitObjectLifetimeEntry HitObjectEntry;
+
+        public readonly JudgementResult Result;
+
+        public JudgementResultEntry(double time, HitObjectLifetimeEntry hitObjectEntry, JudgementResult result)
+        {
+            Time = time;
+            HitObjectEntry = hitObjectEntry;
+            Result = result;
+        }
+
+        public int CompareTo(JudgementResultEntry? other) => Time.CompareTo(other?.Time);
```

Comment:
This has boxing allocation overhead (overload used is `CompareTo(object?)`). With something as important as this maybe important to hand-roll the null check to avoid it.

Reply:
Noticed that this comparator is not actually used.
(stray code from when I was trying to use an ordered set instead of the current stack)